ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -22,31 +22,57 @@
 '''
 
 LOOKAHEAD_WPS = 200 # Number of waypoints we will publish. You can change this number
-
+DEFAULT_VELOCITY = 5 # just picked a number for early tests
 
 class WaypointUpdater(object):
     def __init__(self):
         rospy.init_node('waypoint_updater')
 
-        rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
-        rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        self.subs = {}
+        self.pubs = {}
 
-        # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint below
+        self.subs['/current_pose'] = rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
 
+        self.subs['/base_waypoints'] = rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
 
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
+        self.subs['/current_velocity'] = rospy.Subscriber('/current_velocity', TwistStamped,
+                                                       self.velocity_cb)
+
+        # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint below
+
+        self.pubs['final_waypoints'] = rospy.Publisher('final_waypoints', Lane, queue_size=1)
```

Comment:
May not affect behavior, but should published topic also start with "/"?

Reply:
This was the way it came in the original repo.  Not sure why it is that way - I will test changing it.